File: 03_API_Examples/01_WebAPIGeneral_SampleCode_src/ma.js
async function postRequest(query) {
    const url = 'https://jlp.yahooapis.jp/MAService/V2/parse'
    const req = {
        id: '1',
        jsonrpc: '2.0',
        method: 'jlp.maservice.parse',
        params: { q: query }
    }
    const response = await fetch(url, {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            'User-Agent': 'Yahoo AppID: ' + APPID
        },
        body: JSON.stringify(req)
    })
    return await response.json()
}

async function main() {
    const query = '今日は良い天気です。'
    const response = await postRequest(query)
    console.log(response['result']['tokens'].map(x => x[0] + ':' + x[3]))
}

const APPID = 'あなたの Client ID（アプリケーション ID）';
main()

File: 03_API_Examples/01_WebAPIGeneral_SampleCode_src/ma.py
import json
from urllib import request
APPID = "あなたの Client ID（アプリケーション ID）"
URL = "https://jlp.yahooapis.jp/MAService/V2/parse"
headers = {
    "Content-Type": "application/json",
    "User-Agent": "Yahoo AppID: {}".format(APPID),
}
param_dic = {
    "id": "1234-1",
    "jsonrpc": "2.0",
    "params": {}
}
def ma(query):
    param_dic['params']['q'] = query
    param_dic['method'] = "jlp.maservice.parse"
    params = json.dumps(param_dic).encode()
    req = request.Request(URL, params, headers)
    with request.urlopen(req) as res:
        body = res.read()
    return body.decode()

sentence = '今日は良い天気です。'
response = ma(sentence)
obj = json.loads(response)
tokens = [(x[0], x[3]) for x in obj['result']['tokens']]
print(tokens)
